### PR TITLE
Refactors and gitops improvements

### LINKED ISSUE #11: Update the Software Factory package to do GitOps "correctly"
Currently the Software Factory package (a.k.a. DI2-ME) does the equivalent of `kustomize build | kubectl apply -f -` which uses Flux to deploy everything, but doesn't tell Flux to watch a git repo for changes.

Story:
As a DevOps team maintaining a deployment of the Software Factory package, I want the ability to add commits to a repo in Gitea that holds my deployment configuration and have those changes get automatically picked up by Flux and deployed

Acceptance Criteria:
1. Flux watches a Git repo in Gitea for changes to be applied
2. Airgap compatibility is maintained
3. No manual configuration (CaC only)

### LINKED ISSUE #49: The package only works on Linux
Since we are using `jq` and `kubectl` in the package, but we are pulling in the binaries compiled for linux, the package will only work when run on linux. We should make it usable by any of linux, Mac (intel) and Mac (apple)

### LINKED ISSUE #50: Use a separate Zarf package for Flux
I believe it would be advantageous to use a separate Zarf package for Flux, rather than including it in the main package.

- I see Flux as being on a different level in the "platform hierarchy" than the services that make up the software factory.
- Flux will likely have a different update cycle than the rest of the stack.
- Pulling Flux out will make the main package smaller.
- Increase in operational complexity is minimal, while also reducing the blast radius of changes

### PR DESCRIPTION
**Pull request by RothAndrew**
**Up-front notes:**

* Everything in the `kustomizations/bigbang/vendor` folder is autogenerated and should not be code reviewed. Since Flux doesn't support private bases this is one way to do it. The other way to do it is to modify the Flux deployment to inject git credentials into the pod's `~/.git-credentials` file, but that felt like a worse solution to me than just vendoring the base.

**Changes:**

* Update the package to add the ability for new commits to be pushed to the `zarf-package-software-factory` repo in Gitea and having the changes be automatically reflected in the cluster. This will significantly increase the speed of the feedback loop for development, since we can test changes by pushing commits to Gitea rather than having to re-create a new Zarf Package each time.

* Pull Flux out of the DI2-ME package so that it can be deployed separately with a different lifecycle. This will make the DI2-ME package smaller and prevent the package from mixing "imperative" deployments and "gitops" deployments. With Flux gone the package is now completely gitops

* Makefile updates
    * Add the ability to specify which version of Zarf to use, and pull that version into the local build folder
    * Add the ability to specify which version of Big Bang to use, with documentation on where all the places that need to be updated are when the version of Big Bang changes
    * Add a target for vendoring the Big Bang `/base` folder since Flux doesn't support private bases
    * Add a target for creating the Zarf Package for Flux

* Add to Known Issues section of README -- Jenkins won't work in disconnected environments due to its dependency on plugins pulled from the internet

* Increase CPUs in Vagrantfile -- The old value wasn't big enough even for just Istio and GitLab with everything else disabled

Fixes #11 
Fixes #50 
Fixes #49

### CODE REVIEW

**Comment by RothAndrew** [comment]
image

image

image

image

image

image

image

image

image

image

image

image

image

image

**Comment by RothAndrew** [comment]
Last change to be made is to change the branch referenced in the `GitRepo` resources from this branch to `main`. Will make that change right before merging.

File: zarf.yaml [hunk 1]
```diff
@@ -166,47 +135,69 @@ components:
...
-      - kiwigrid/k8s-sidecar:1.14.2
       - jenkins/inbound-agent:4.11-1
-      - maorfr/kube-tasks:0.2.0
+      - kiwigrid/k8s-sidecar:1.14.2
+
+
+
+
+
+#  - name: atlassian
```
Comment (YrrepNoj) [inline zarf.yaml:160]: Lots of commented out configuration below that seems to be (mostly?) replaced above. Is any of this still being worked or can it just be removed now?

Comment (RothAndrew) [reply zarf.yaml:160]: oops, that's leftover. will remove

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**